### DISCUSSION TITLE
en/tutorials/redirect/

### DISCUSSION THREAD

**Discussion by giscus** [go-gost/gost#502] [General]
# en/tutorials/redirect/

https://gost.run/en/tutorials/redirect/

**Comment by Zensey** [comment]
Hi
This doesn't seem to work for docker containers, programs inside container loose connection with internet. I mean TCP protocol.

**Comment by Zensey** [comment]
Traffic from inside Docker container does not reach the proxy server, however the packet counter of the relevant iptables route increments.

**Reply by lemonc9** [reply]
I also encountered this problem, do you have a solution?

**Comment by aogg** [comment]
用了gost -L "red://:12345?sniffing=true&tproxy=true&so_mark=100"
到底还需不需要额外的输入iptables -t nat -A OUTPUT -p tcp --match multiport ! --dports 12345,8899 -j DNAT --to-destination 127.0.0.1:12345

**Comment by maddo7** [comment]
I want to run all trafic from my android phone A through my other android phone B. both phones are inthe same wifi network.
Both phones are rooted so I though about using ghost tunnel for it.

so I first pasted all the TCP rules 1:1 into the shell of phone A, then all UDP rules 1:1 from the tutorial.
192.168.50.8 is the ip in my network of phone B.

Then I started on phone A two gost tunnels according to the tutorial:
```
./gost -L "red://:12345?sniffing=true&tproxy=true" -F "relay://192.168.50.8:8001?so_mark=100"
./gost -L "redu://:12345?ttl=30s" -F "relay://192.168.50.8:8001?so_mark=100"
```

on phone B I started one
```
./gost -L "relay://:8001?interface=rmnet_data0&resolver=1.1.1.1,tcp://8.8.8.8,tls://8.8.8.8:853,https://1.0.0.1/dns-query"
```

now if I use curl on phone A through the console, it shows the mobile IP of phone B. If I kill the gost process on phone B, curl will time out and not work.
However, the webbrowser on phone A still seems to ignore this. When I go to an IP showing website, it shows the IP address of the local network and if I kill the gost process on phone B, the browser still works.
Why is the browser ignoring the settings and how can I make this work so that all traffic goes through phone B?

**Comment by killgxlin** [comment]
iptables -t mangle -A GOST_LOCAL -p tcp -j MARK --set-mark 1
这行为什么没有使用TPROXY呢？没有标记12345端口，red可以收到这个包吗？
